### PR TITLE
Update `FilePath::getFilesInDirectory` to allow getting all files

### PR DESCRIPTION
**Pull request by ld-kerley**
While working on some other internal projects I found myself using the `FilePath` object, and bumped up against the fact that `getFilesInDirectory` is unable to query all the files in the provided directory.

This PR provides a simple extension to the existing code to allow accessing all the files in a directory if no extension is provided to match against.

NOTE: I am not a Windows developer, so I would greatly appreciate someone more familiar with the Windows API checking what I'm proposing here will work.

### CODE REVIEW

File: source/MaterialXFormat/File.h [hunk 1]
```diff
@@ -188,7 +188,8 @@ class MX_FORMAT_API FilePath
     bool isDirectory() const;
 
     /// Return a vector of all files in the given directory with the given extension.
-    FilePathVec getFilesInDirectory(const string& extension) const;
+    /// If extension is empty all files are returned.
+    FilePathVec getFilesInDirectory(const string& extension="") const;
```
Comment (kwokcb) [inline source/MaterialXFormat/File.h]: `extension = EMPTY_STRING` instead to follow coding guidelines :)

**Review by kwokcb — COMMENTED** [review]
Maybe add a unit test for empty string arg? 
That's how I test for Linux since I don't have that.
